app/index.js

```diff
@@ -44,7 +44,10 @@ module.exports = generators.Base.extend({
         type: 'input',
         name: 'app_id',
         message: 'App ID',
-        default: answers => answers.name
+        validate: (input) => {
+          return (/\-|\s/.test(input)) ? 'Invalid App ID, use alphanumeric and periods only' : true;
```

Comment:
This regex only tests for dashes and spaces. All other characters like `#*/&!~` etc. would slip through. If the app id is tested, it should be a positive test such as `/[a-z0-9.]+/`